org.eclipse.lsp4e.debug/src/org/eclipse/lsp4e/debug/launcher/DSPLaunchDelegate.java

```diff
@@ -384,8 +384,26 @@ public void close() {
 	 * of {@link DSPDebugTarget}, but does not have to be. The arguments to this
 	 * method are normally just passed to {@link DSPDebugTarget} constructor.
 	 */
-	protected IDebugTarget createDebugTarget(SubMonitor subMonitor, Supplier<TransportStreams> streamsSupplier, ILaunch launch, Map<String, Object> dspParameters) throws CoreException {
-		final var target = new DSPDebugTarget(launch, streamsSupplier, dspParameters);
+	protected IDebugTarget createDebugTarget(SubMonitor subMonitor, Supplier<TransportStreams> streamsSupplier,
+			ILaunch launch, Map<String, Object> dspParameters) throws CoreException {
+		/*
+		 * TODO use the following code when removing createDebugTarget deprecated
+		 * final var target = new DSPDebugTarget(launch, streamsSupplier,
+		 * dspParameters); target.initialize(subMonitor.split(80)); return target;
+		 * return target;
+		 */
+		final var supplier = streamsSupplier.get();
+		return createDebugTarget(subMonitor, supplier::close, supplier.in, supplier.out, launch, dspParameters);
```

Comment:
The newer/recommended method should not reference the deprecated one, it would better be the other way round.

Reply:
That does not work since the new `createDebugTarget` method is invoked by `public void launch(DSPLaunchDelegateLaunchBuilder builderSrc)` of the same class. I would then have to revert all your changes to this class. 

If I make the deprecated method delegate to the new method it will effectively never be called, thus overriding it will be without any effect.